african-market-place/src/components/pages/Login/index.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+
+export const LoginPage = () => {
+  const initialForm = {
+    username: '',
+    password: '',
+  };
+
+  const [form, setForm] = useState(initialForm);
+
+  const onChange = e => {
+    const { name, value } = e.target;
+    setForm({
+      ...form,
+      [name]: value,
+    });
+  };
```

Comment:
Does this work? wouldnt setting e.target to name anmd value in line 12 just make line 15 be [e.target]:e.target not [e.target.name]:e.target,value

Reply:
Yeah this works.  It will make whatever 'name' is, the property on the state.